### ISSUE TITLE
Adaptive rest

### ISSUE THREAD

**Issue by brandonp2412** [brandonp2412/Flexify#74]
Rest times could be adjusted based on a formula estimating how hard a set was. My thinking was either:

1. One rep max estimate loss
2. Rep loss
3. Volume loss
4. Reps per minute loss

By loss I mean the difference between the last two sets. E.g. imagine a person last week does:

1:10pm 10x60kg Bench press (warmup ignoring)
1:15pm 4x80kg Bench press
1:20pm 3x80kg   Bench press
1:30pm 2x80kg   Bench press

The goal would be to calculate how tired the person is getting from each set, and increase the rest duration (within reason) to adjust for their fatigue.

1. 87.3kg -> 84.7kg -> 82.3kg
2. 4 -> 3 -> 2
3. 320kg -> 240kg -> 160kg
4. 0.8 -> 0.6 -> 0.4

The simplest and probably about as accurate as the rest would be rep loss, so we could say something like:
`restDuration * (1 - repLoss)`

**Comment by Crazy-Unicooorn** [comment]
Wouldn't that formula decrease rest time? I think you meant `restDuration / (1 - repLoss)`, or it's not clear to me what repLoss represents 😄

But I believe dividing is risky…

What if we calculate like this :
repLoss = previousVol / currentVol
adjusted restDuration = `restDuration * repLoss`

so in your example we have
320 kg > 240 kg > 160 kg
So from 320 to 240 we have 320 / 240 = 1.33 and then 240 / 160 = 1.5

If I have a restDuration of 2 min, that results in:

2 * 1.33 = 2.66 min
and
2 * 1.5 = 3 min

I think it makes sense 🤔

---

What about sets without weight?

Let's say I do 20 pushups, then 18, then 14
Could we adjust restDuration still?

2 * (20 / 18) = 2.22
2 * (18 / 14) = 2.57

That's just food for thoughts, not sure if my formulas are accurate

**Comment by brandonp2412** [comment]
Yeah keen to just make something and have it as a setting turned off, then we can just test out if its working or not. Rep estimation I did similar stuff and it seems to be almost correct for me.

**Comment by brandonp2412** [comment]
Not feeling it. The more I work on trying to estimate rest timers the more it feels like I run in circles.
